src/utils/index.js

```diff
@@ -36,8 +38,15 @@ var writeErrorHead = function( res, statusCode ) {
  */
 var fourOhFour = function( res, msg, url ) {
   log.warn( msg + ': ' + ( ( url != null ? url.format() : void 0 ) || 'unknown' ) );
-  writeErrorHead( res, 404 );
-  return finish( res, 'Not Found' );
+
+  try {
+    var data = fs.readFileSync('./placeholders/404.jpg');
```

Comment:
blocking file i/o will prevent kamu from responding to other requests in parallel